src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -331,27 +330,32 @@ private async Task ReconnectAsync(CancellationToken cancellationToken)
                 // establishes the first connection, throws if it fails
                 await ConnectFirstAsync(cancellationToken).CfAwait();
 
-                // FIXME reconnect, shall we wait for the first member view event?
-                // else we are just hoping that we haven't switched to a new
-                // cluster and that this connection matches a member?
+                // once the first connection is established, we should use it to subscribe
+                // to the cluster views event, and then we should receive a members view,
+                // which in turn should change the state to Connected - unless something
+                // goes wrong
+                var connected = await _clusterState.WaitForConnectedAsync(cancellationToken).CfAwait();
+
+                if (!connected)
+                {
+                    // we are a background task and cannot throw!
+                    _logger.LogError("Failed to reconnect.");
+                }
 
-                // and now we have been re-connected (rejoice)
-                // though nothing guarantees we still are, or will remain for long
-                // but OnConnectionClosed will deal with it
+                // we have been reconnected (rejoice) - of course, nothing guarantees that it
+                // will last, but then OnConnectionClosed will deal with it
             }
             catch (Exception e)
             {
-                // we *have* retried and failed
-                await _clusterState.TransitionAsync(ClientState.Shutdown).CfAwait();
-
-                // we are a background task and cannot throw!
+                // we *have* retried and failed, shutdown, and log (we are a background task!)
+                _clusterState.RequestShutdown();
                 _logger.LogError(e, "Failed to reconnect.");
             }
 
             // in any case, remove ourselves
             _reconnect = null;
         }
-
+        
```

Comment:
```suggestion

```

Reply:
Don't worry too much about these. There will be another PR that essentially clears all these rogue spaces (it's done with an automatic tool).